internal/cli/universal_login_templates.go

```diff
@@ -0,0 +1,454 @@
+)
+
+const (
+	defaultPrimaryColor    = "#0059d6"
+	defaultBackgroundColor = "#000000"
+	defaultLogoURL         = "https://cdn.auth0.com/manhattan/versions/1.3935.0/assets/badge.png"
+)
+
+var (
+	errNotAllowed = errors.New(
+		"this feature requires at least one custom domain to be set and verified " +
+			"for the tenant, use 'auth0 domains create' to create one and 'auth0 domains verify' to have it verified",
+	)
+
+	//go:embed data/branding/storybook/*
+	templatePreviewAssets embed.FS
+
+	//go:embed data/branding/tenant-data.js
+	tenantDataAsset string
+
+	//go:embed data/branding/default-template.liquid
+	templateBasic string
+
+	//go:embed data/branding/image-template.liquid
+	templateWithImage string
+
+	//go:embed data/branding/footer-template.liquid
+	templateWithFooter string
+
+	templateBody = Flag{
+		Name:       "Template",
+		LongForm:   "template",
+		ShortForm:  "t",
+		Help:       "Custom page template for Universal Login.",
+		IsRequired: true,
+	}
+
+	templateOptions = pickerOptions{
+		{"Basic template", templateBasic},
+		{"Template with login box and background image", templateWithImage},
+		{"Template with page footers", templateWithFooter},
+	}
+)
+
+// TemplateData contains all the variables we project onto our embedded go
+// template. These variables largely resemble the same ones in the auth0
+// branding template.
+type TemplateData struct {
+	Filename        string
+	Clients         []ClientData
+	PrimaryColor    string
+	BackgroundColor string
+	LogoURL         string
+	TenantName      string
+	Body            string
+}
+
+// ClientData is a minimal representation of an Auth0 Client as defined in the
+// management API. This is used within the branding machinery to populate the
+// tenant data.
+type ClientData struct {
+	ID      string `json:"id"`
+	Name    string `json:"name"`
+	LogoURL string `json:"logo_url,omitempty"`
+}
+
+func showBrandingTemplateCmd(cli *cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "show",
+		Args:  cobra.NoArgs,
+		Short: "Display the custom template for Universal Login",
+		Long:  "Display the custom template for the Universal Login experience.",
+		Example: `  auth0 universal-login templates show
+  auth0 ul templates show`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var currentTemplate *management.BrandingUniversalLogin
+			if err := ansi.Waiting(func() (err error) {
+				currentTemplate, err = cli.api.Branding.UniversalLogin()
+				if err != nil {
+					if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+						return nil
+					}
+					return err
+				}
+
+				return nil
+			}); err != nil {
+				return fmt.Errorf("failed to load the Universal Login template: %w", err)
+			}
+
+			cli.renderer.Heading("universal login template")
+
+			if currentTemplate == nil {
+				cli.renderer.Infof("No custom template found. To set one, run: `auth0 universal-login templates update`.")
+			}
+
+			fmt.Println(currentTemplate.GetBody())
```

Comment:
Should we have this in a Renderer func, inside the `display` package as usual?

Reply:
We actually already do `cli.renderer.Output(currentTemplate.GetBody())` this was just moved but I can update it.

Reply:
[`342a03a` (#666)](https://github.com/auth0/auth0-cli/pull/666/commits/342a03ae50dfde08b5581a5f410004aa7ee0b5c0)

Reply:
I meant in a separate file inside the `display` package, as in:

<img width="726" alt="Screenshot 2023-03-10 at 14 30 26" src="https://user-images.githubusercontent.com/5055789/224383869-06e1993a-d7af-4539-bb0d-ef757ae6166d.png">

But it's a nit anyway, nevermind.